chain/pra/receiver.go

```diff
@@ -84,6 +85,9 @@ func (r *Receiver) newReceiptProofs(v *BlockNotification) ([]*chain.ReceiptProof
 	rps := make([]*chain.ReceiptProof, 0)
```

Comment:
@trmaphi I suggest a refactor for this function like this

```
func (r *Receiver) newReceiptProofs(v *BlockNotification) ([]*chain.ReceiptProof, error) {
	rps := make([]*chain.ReceiptProof, 0)

	if len(v.Events.EVM_Log) > 0 {
		rp := &chain.ReceiptProof{
			Height: int64(v.Height),
		}

		if key, proofs, err := r.getProofs(v); err != nil {
			return nil, err
		} else if rp.Proof, err = codec.RLP.MarshalToBytes(&StateProof{
			Key:   key,
			Value: proofs,
		}); err != nil {
			return nil, err
		}

		foundMessageEvent := false
		for _, e := range v.Events.EVM_Log {
			if !strings.EqualFold(e.Log.Address.Hex(), r.src.ContractAddress()) {
				continue
			}

			if bmcmsg, err := r.c.bmc.ParseMessage(e.EvmLog()); err != nil {
				// TODO: handle this error in a right way.
				// if this error must not happend. We just stop the app right here.
				r.l.Warnf("failed to parse message err:%v", err)
				continue
			} else {
				foundMessageEvent = true
				if bmcmsg.Seq.Int64() == int64(r.rxSeq) {
					r.isFoundMessageEventByOffset = true
				}
				rp.Events = append(rp.Events, &chain.Event{
					Message:  bmcmsg.Msg,
					Next:     chain.BtpAddress(bmcmsg.Next),
					Sequence: bmcmsg.Seq.Int64(),
				})
			}
		}

		if foundMessageEvent {
			rps = append(rps, rp)
		}
	}

	return rps, nil
}
```

Reply:
BMC can emit multiple event